--- post.inc.php.orig	2008-04-06 20:31:33.000000000 +0200
+++ post.inc.php	2008-04-06 20:34:27.000000000 +0200
@@ -3,9 +3,12 @@
 // All tests which apply specifically to POST requests
 function bb2_post($settings, $package)
 {
-	// Check blackhole lists for known spam/malicious activity
-	require_once(BB2_CORE . "/blackhole.inc.php");
-	bb2_test($settings, $package, bb2_blackhole($package));
+    // config check added for DokuWiki plugin
+    if(!$settings['skipblackhole']){
+    	// Check blackhole lists for known spam/malicious activity
+    	require_once(BB2_CORE . "/blackhole.inc.php");
+    	bb2_test($settings, $package, bb2_blackhole($package));
+    }
 
 	// MovableType needs specialized screening
 	if (stripos($package['headers_mixed']['User-Agent'], "MovableType") !== FALSE) {
