--- core.inc.php.orig	2011-04-09 17:06:01.000000000 +0200
+++ core.inc.php	2011-04-09 17:33:09.000000000 +0200
@@ -144,11 +144,14 @@
 		bb2_test($settings, $package, bb2_blacklist($package));
 
 		// Check the http:BL
-		require_once(BB2_CORE . "/blackhole.inc.php");
-		if (bb2_test($settings, $package, bb2_httpbl($settings, $package))) {
-			// Bypass all checks if http:BL says search engine
-			bb2_approved($settings, $package);
-			return true;
+		// config check added for DokuWiki plugin
+		if(!$settings['skipblackhole']){
+			require_once(BB2_CORE . "/blackhole.inc.php");
+			if (bb2_test($settings, $package, bb2_httpbl($settings, $package))) {
+				// Bypass all checks if http:BL says search engine
+				bb2_approved($settings, $package);
+				return true;
+			}
 		}
 
 		// Check for common stuff
